### PR TITLE
feat: use max_results in SDK's list_trials

### PR DESCRIPTION
**Pull request by azhou-determined**
## Description

change list_trials to take in max_results instead of page limit

## Test Plan

ensure that list_trials with max_results returns specified number of trials
```
exp = client.get_experiment(10)
exp.list_trials(max_results=2)
```
## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/tests/determined/common/experimental/test_experiment.py [hunk 1]
```diff
@@ -101,7 +100,7 @@ def test_wait_raises_exception_when_experiment_is_paused(
 
 
 @responses.activate
-def test_list_trials_iterates_through_all_trials(
```
Comment (wes-turner) [inline harness/tests/determined/common/experimental/test_experiment.py:104]: I like the new test. Why did you delete the old ones? They still look relevant to me.

Comment (wes-turner) [reply harness/tests/determined/common/experimental/test_experiment.py:104]: small nit: in case you prefer the language, `test_list_trials_respects_max_results` seems a little more accurate to me. But I could also understand if you think it obscures some of the test's meaning, too.

Comment (azhou-determined) [reply harness/tests/determined/common/experimental/test_experiment.py:104]: > Why did you delete the old ones?

goofed. added it back.

File: harness/tests/determined/common/experimental/test_experiment.py [hunk 2]
```diff
@@ -143,7 +143,7 @@ def test_list_trials_requests_pages_lazily(
         callback=api_responses.serve_by_page(tr_resp, "trials", max_page_size=page_size),
     )
 
-    trials = expref.list_trials(limit=page_size)
+    trials = expref.list_trials()
```
Comment (wes-turner) [inline harness/tests/determined/common/experimental/test_experiment.py:146]: Nice. This seems better, anyway.

File: harness/tests/determined/common/experimental/test_experiment.py (hunk 1 above)
Comment (wes-turner) [reply harness/tests/determined/common/experimental/test_experiment.py:104]: `test_list_trials_respects_max_results` is a good addition, but I think I expect `test_list_trials_iterates_through_all_trials` to still be a valid test for when `max_results` isn't passed. Agree/disagree?

Comment (azhou-determined) [reply harness/tests/determined/common/experimental/test_experiment.py:104]: makes sense. i should have treated `max_results` as a new feature, these tests are still valid.

**Review by wes-turner — APPROVED** [review]
LGTM
